### PR TITLE
550 724 add create schema button

### LINKED ISSUE #550: Add "Create Schema" button in Schemas view empty state when Schema Registry is selected
Similar to how we have "Create Topic" in the Topics view empty state when a Kafka cluster is selected:
https://github.com/confluentinc/vscode/blob/33b7a8e63f3b37427d64f04edbb5bd84e4f34b73/package.json#L433-L437

image

Should:
 1. Quickpick ask what flavor schema they want (avro, pbuf, json)
 2. Create the empty buffer with right language type / extension
 3. Record what schema registry this file URI should be for,
 4. Ensure that the 'ccloud upload' icon is active for the buffer, then when hit ...
 5. Should skip the schema registry quickpick 'cause we knew it already, and go straight to the 'pick a subject' step.
 6. ...
 7. Profit!

When that flow is working, then can also add a '+' button to the top bar of the schemas view when a schema registry has been chosen that drives the same path, parallel to the one driving 'create new Kafka topic.'

Duplicated by #724 , implement the best union of the two.

### LINKED ISSUE #724: CTA to create new schema (possibly from scratch)
Bind a '+' button action to the top of the schema registry view when a registry has been selected. Tooltip 'Create New Schema'. It should:

  1. Quickpick ask the user what language (Avro, Proto, JSON Schema)
  2. Ask for a filename / path to save it as, defaulting to the corresponding file extension. Use the vscode save file chooser functionality.
  3. If user picked an existing file, open it, otherwise create new file from scratch, then open. Encode both the schema registry id and the schema langauge in the query string.
  4. Set the right language on the buffer based on the originally chosen language union their installed language ids.
 
 If done right, then the 'cloud upload' icon should be present in upper-right hand of editor, and any installed language syntax helper should kick in while editing (since is a file: scheme document).
 
 Duplicates #550 , implement the best union of the two.

### PR DESCRIPTION
**Pull request by jlrobins**
## Summary of Changes

- "Create new schema" invitation button from the empty state of a schema registry.

## Any additional details or context that should be provided?

- Does not do any magic encoding of information into the new document's URI. We have a plan to do away with all of that.
- Only adds the CTA in the empty schema registry state. Future work will also add a CTA as an icon on the schema registry top row itself.
- Closes #550. Closes #724.

image

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [ ] Added new
- [ ] Updated existing
- [ ] Deleted existing

##### Other

- [ ] All new disposables (event listeners, views, channels, etc.) collected as  for eventual cleanup?

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [ ] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: src/commands/schemas.ts [hunk 1]
```diff
@@ -266,22 +292,24 @@ async function loadOrCreateSchemaViewer(schema: Schema): Promise<vscode.TextEdit
  * Possibly set the language of the editor's document based on the schema.
  * Depends on what languages the user has installed.
  */
-async function setEditorLanguageForSchema(textDoc: vscode.TextEditor, schema: Schema) {
+async function setEditorLanguageForSchema(textDoc: vscode.TextEditor, type: SchemaType) {
```
Comment (jlrobins) [inline src/commands/schemas.ts:295]: Needed to make work independent of an entire Schema object.

File: src/models/schema.ts [hunk 2]
```diff
@@ -23,9 +20,13 @@ const extensionMap: { [key in SchemaType]: string } = {
 const languageTypes: { [key in SchemaType]: string[] } = {
   [SchemaType.Avro]: ["avroavsc", "json"],
   [SchemaType.Json]: ["json"],
-  [SchemaType.Protobuf]: ["proto"],
+  [SchemaType.Protobuf]: ["proto3", "proto"],
 };
 
+export function getLanguageTypes(schemaType: SchemaType): string[] {
```
Comment (jlrobins) [inline src/models/schema.ts:26]: Refactor to be useable w/o an entire Schema object.

File: src/commands/schemas.ts [hunk 3]
```diff
@@ -76,6 +78,30 @@ function validateCommand(item: any) {
...
 }
 
+/** User has gestured to create a new schema from scratch relative to the currently selected schema registry. */
+async function createSchemaCommand() {
+  const schemaRegistry: SchemaRegistry | null = getSchemasViewProvider().schemaRegistry;
+
+  if (!schemaRegistry) {
+    logger.error("createSchemaCommand called with no schema registry selected");
+    return;
+  }
```
Comment (shouples) [inline src/commands/schemas.ts:88]: We could probably remove this, yeah?

**Review by shouples — APPROVED** [review]
Good stuff!

**Pull request merged**